cmd/ftl/cmd_deploy.go

```diff
@@ -100,10 +103,54 @@ func (d *deployCmd) Run(ctx context.Context, client ftlv1connect.ControllerServi
 		return err
 	}
 
+	if d.Wait {
+		logger.Infof("Waiting for deployment %s to become ready", resp.Msg.DeploymentName)
+
+		var wg *errgroup.Group
+		wg, ctx = errgroup.WithContext(ctx)
+		wg.Go(d.checkReadiness(ctx, client, resp.Msg.DeploymentName))
+		if err := wg.Wait(); err != nil {
+			return fmt.Errorf("deployment %s failed to become ready: %w", resp.Msg.DeploymentName, err)
+		}
+	}
+
 	logger.Infof("Successfully created deployment %s", resp.Msg.DeploymentName)
 	return nil
 }
 
+func (d *deployCmd) checkReadiness(ctx context.Context, client ftlv1connect.ControllerServiceClient, deploymentName string) func() error {
+	return func() error {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				status, err := client.Status(ctx, connect.NewRequest(&ftlv1.StatusRequest{
+					AllDeployments: true,
+				}))
+				if err != nil {
+					return err
+				}
+
+				var found bool
+				for _, deployment := range status.Msg.Deployments {
+					if deployment.Key == deploymentName {
+						found = true
+						if deployment.Replicas >= d.Replicas {
+							return nil
+						}
+					}
+				}
+				if !found {
+					return fmt.Errorf("deployment %s not found: %v", deploymentName, status.Msg.Deployments)
```

Comment:
Is the assumption here that the deployment should already exist? Is that true?

Reply:
it should be true, we're performing the `wait` loop synchronously after calling the controller to persist the deployment via `client.CreateDeployment`